### PR TITLE
bugfix/modifying Regex for db snapshot identifier

### PR DESCRIPTION
**Pull request by Rohit1509**
Signed-off-by: Rohit Joshi <rohit.prasad.joshi@sap.com>

### Description

Please describe what this change achieves. Ensure you have read the [Contributing to InSpec AWS](https://github.com/inspec/inspec-aws/CONTRIBUTING.md) document before submitting.

### Issues Resolved

List any existing issues this PR resolves, or any Discourse or StackOverflow discussion that's relevant

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [ ] New functionality includes integration tests/controls
- [ ] New Terraform resources
- [ ] Documentation provided or updated for resources 
- [ ] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Comment by Rohit1509** [comment]
The system generated snapshots contain a colon(:) in their ID. Hence modifying the regex in libraries accordingly.

**Review by soumyo13 — APPROVED** [review]
LGTM

**Comment by sathish-progress** [comment]
> The system generated snapshots contain a colon(:) in their ID. Hence modifying the regex in libraries accordingly.

@Rohit1509  As per the [docs](https://docs.aws.amazon.com/cli/latest/reference/rds/create-db-snapshot.html#options), it is supposed to not have colon ..

Also I did try adding a snapshot with a colon in it and it fails.. while the regex you added allows it .. Could you change that please?
Screenshot 2021-09-13 at 10 56 57 AM

File: libraries/aws_rds_snapshot.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@ def initialize(opts = {})
     super(opts)
     validate_parameters(required: [:db_snapshot_identifier])
 
-    raise ArgumentError, "#{@__resource_name__}: db_snapshot_identifier must start with a letter followed by up to 62 letters/numbers/hyphens." if opts[:db_snapshot_identifier] !~ /^[a-z]{1}[0-9a-z\-]{0,62}$/
+    raise ArgumentError, "#{@__resource_name__}: db_snapshot_identifier must start with a letter followed by up to 1 to 255 characters." if opts[:db_snapshot_identifier] !~ /^[a-z]{1}[0-9a-z\-\:]{1,255}$/
```
Comment (sathish-progress) [inline libraries/aws_rds_snapshot.rb:20]: please review the recommendation from here https://github.com/inspec/inspec-aws/pull/622#issuecomment-917853734

**Comment by Rohit1509** [comment]
> > The system generated snapshots contain a colon(:) in their ID. Hence modifying the regex in libraries accordingly.
> 
> @Rohit1509 As per the [docs](https://docs.aws.amazon.com/cli/latest/reference/rds/create-db-snapshot.html#options), it is supposed to not have colon ..
> 
> Also I did try adding a snapshot with a colon in it and it fails.. while the regex you added allows it .. Could you change that please?
> Screenshot 2021-09-13 at 10 56 57 AM

Here is the example of system generated snapshot with a colon.
rds2

How can we deal with these ones ?

**Comment by sathish-progress** [comment]
> > > The system generated snapshots contain a colon(:) in their ID. Hence modifying the regex in libraries accordingly.
> > 
> > 
> > @Rohit1509 As per the [docs](https://docs.aws.amazon.com/cli/latest/reference/rds/create-db-snapshot.html#options), it is supposed to not have colon ..
> > Also I did try adding a snapshot with a colon in it and it fails.. while the regex you added allows it .. Could you change that please?
> > Screenshot 2021-09-13 at 10 56 57 AM
> 
> Here is the example of system generated snapshot with a colon.
> rds2
> 
> How can we deal with these ones ? @sathish-progress

Okay let me check this and get back again..

**Comment by sathish-progress** [comment]
> > > > The system generated snapshots contain a colon(:) in their ID. Hence modifying the regex in libraries accordingly.
> > > 
> > > 
> > > @Rohit1509 As per the [docs](https://docs.aws.amazon.com/cli/latest/reference/rds/create-db-snapshot.html#options), it is supposed to not have colon ..
> > > Also I did try adding a snapshot with a colon in it and it fails.. while the regex you added allows it .. Could you change that please?
> > > Screenshot 2021-09-13 at 10 56 57 AM
> > 
> > 
> > Here is the example of system generated snapshot with a colon.
> > rds2
> > How can we deal with these ones ? @sathish-progress
> 
> Okay let me check this and get back again..

@Rohit1509 you are right about the colon and inspec rds_snapshot resource needs to allow such identifiers

File: libraries/aws_rds_snapshot.rb (hunk 1 above)
Comment (sathish-progress) [reply libraries/aws_rds_snapshot.rb:20]: this is needed to support system generated snapshots.

**Review by sathish-progress — APPROVED** [review]
LGTM

**Pull request merged**